docs/DifferentiableFunctions.md

```diff
@@ -431,8 +431,8 @@ defined:
 // functions. It simply returns `pullback(1)`.
 func gradient<T, R>(
     at x: T, in f: @differentiable (T) -> R
-) -> T.CotangentVector
-    where T: Differentiable, R: FloatingPoint & Differentiable, R.CotangentVector == R
+) -> T.TangentVector
+    where T: Differentiable, R: FloatingPoint & Differentiable, R.TangentVector == R
```

Comment:
```suggestion
    where R: FloatingPoint, R.TangentVector == R
```

Reply:
I deliberately didn’t apply this change in this PR since this is about removing cotan, and I think there should be a paragraph describing the requirement inference behavior.

Reply:
I see, that makes sense.